drivers/vulkan/rendering_device_vulkan.cpp

```diff
@@ -9027,6 +9026,8 @@ void RenderingDeviceVulkan::capture_timestamp(const String &p_name) {
 		vkCmdPipelineBarrier(frames[frame].draw_command_buffer, VK_PIPELINE_STAGE_ALL_COMMANDS_BIT, VK_PIPELINE_STAGE_ALL_COMMANDS_BIT, 0, 1, &memoryBarrier, 0, nullptr, 0, nullptr);
 	}
 
+	//Reset all queries in a query pool before writing a timestamp. This is according to the Vulkan spec.
+	vkCmdResetQueryPool(frames[frame].setup_command_buffer, frames[frame].timestamp_pool, 0, max_timestamp_query_elements);
```

Comment:
It looks a bit fishy to submit the reset on the setup command buffer, but write the timestamps using the draw_command_buffer. Is there are reason that these operations happen on different command buffers?

Reply:
If we would reset a queue pool with draw_command_buffer then frame time view would brake. 
![image](https://github.com/godotengine/godot/assets/33091666/199fa3ed-57f3-41cd-a8a4-a3f615798585)

Can't pinpoint what's the underlying reason to it.

Reply:
It does seem strange but maybe it's a timing thing? It would be nice if we could dig up some more info on why this needs to be so

Reply:
It kind of makes sense as the setup command buffer is queued first, so the commands in it run first. But I feel like we shouldn't need to reset the entire query pool each time we capture a single timestamp

Looking at the old code, it seems like we only reset the queries that were used in the last frame. So if each frame we use a different number of queries, then in frames where we use more, we will be using queries that were not reset. Maybe it would be enough if instead of resetting only ``frames[frame].timestamp_count`` of queries we reset all queries at the beginning of the frame?

Reply:
Makes sense, the entire query pool needs to be reset once. At least validation errors spawned only the first time when the user opens the frame view.

Reply:
@BastiaanOlij @clayjohn updated how resetting query pools work. Now we just reset entire query pools only once and after that we reset them how we did before this pr.